compiler/luci-interpreter/pal/cmsisnn/PALDepthwiseConv2d.h

```diff
@@ -0,0 +1,137 @@
+    (void)output_data;
+    (void)input_data;
+    (void)filter_shape;
+    (void)filter_data;
+    (void)bias_shape;
+    (void)bias_data;
+    (void)output_shape;
+    (void)output_data;
+  }
+}
+
+template <>
+inline void
+DepthwiseConvPerChannel<int8_t>(const tflite::DepthwiseParams &params,
+                                const int32_t *output_multiplier, const int32_t *output_shift,
+                                const tflite::RuntimeShape &input_shape, const int8_t *input_data,
+                                const tflite::RuntimeShape &filter_shape, const int8_t *filter_data,
+                                const tflite::RuntimeShape &bias_shape, const int32_t *bias_data,
+                                const tflite::RuntimeShape &output_shape, int8_t *output_data)
+{
+  cmsis_nn_dw_conv_params dw_conv_params;
+  dw_conv_params.dilation.h = params.dilation_height_factor;
+  dw_conv_params.dilation.w = params.dilation_width_factor;
+
+  if (dw_conv_params.dilation.h == 1 && dw_conv_params.dilation.w == 1)
+  {
+    dw_conv_params.input_offset = params.input_offset;
+    dw_conv_params.output_offset = params.output_offset;
+    dw_conv_params.stride.h = params.stride_height;
+    dw_conv_params.stride.w = params.stride_width;
+    dw_conv_params.padding.h = params.padding_values.height;
+    dw_conv_params.padding.w = params.padding_values.width;
+
+    dw_conv_params.activation.min = params.quantized_activation_min;
+    dw_conv_params.activation.max = params.quantized_activation_max;
+    dw_conv_params.ch_mult = params.depth_multiplier;
+
+    cmsis_nn_per_channel_quant_params quant_params;
+    int32_t output_multiplier = params.output_multiplier;
+    int32_t output_shift = params.output_shift;
+
+    quant_params.multiplier = &output_multiplier;
+    quant_params.shift = &output_shift;
+
+    const int batch_size = tflite::MatchingDim(input_shape, 0, output_shape, 0);
+    const int output_depth = tflite::MatchingDim(filter_shape, 3, output_shape, 3);
+
+    cmsis_nn_dims input_dims;
+    input_dims.n = batch_size;
+    input_dims.h = input_shape.Dims(1);
+    input_dims.w = input_shape.Dims(2);
+    input_dims.c = input_shape.Dims(3);
+
+    cmsis_nn_dims filter_dims;
+    filter_dims.n = filter_shape.Dims(0);
+    filter_dims.h = filter_shape.Dims(1);
+    filter_dims.w = filter_shape.Dims(2);
+    filter_dims.c = output_depth;
+
+    cmsis_nn_dims bias_dims;
+    bias_dims.n = 1;
+    bias_dims.h = 1;
+    bias_dims.w = 1;
+    bias_dims.c = output_depth;
+
+    cmsis_nn_dims output_dims;
+    output_dims.n = batch_size;
+    output_dims.h = output_shape.Dims(1);
+    output_dims.w = output_shape.Dims(2);
+    output_dims.c = output_depth;
+
+    cmsis_nn_context ctx;
+    ctx.buf = nullptr;
+    ctx.size = 0;
+
+    const int32_t buf_size = arm_depthwise_conv_wrapper_s8_get_buffer_size(
+      &dw_conv_params, &input_dims, &filter_dims, &output_dims);
+
+    ctx.buf = malloc(buf_size);
+    ctx.size = buf_size;
```

Comment:
1. Why we can't to define `buf_size` first and then create cmsis_nn_context?
2. Could we avoid usage of malloc and use `std::make_unique<int8_t[]>(buf_size);`?
3. If `malloc` is neccessary, please, add some assert.
4. It's Okay to allocate memory without introduced memory allocator?

```suggestion
    const int32_t buf_size = arm_depthwise_conv_wrapper_s8_get_buffer_size(
      &dw_conv_params, &input_dims, &filter_dims, &output_dims);
    auto buffer = std::make_unique<int8_t[]>(buf_size);
    
    cmsis_nn_context ctx;
    {
      ctx.buf = buffer.get();
      assert(ctx.buf != nullptr);
      ctx.size = buf_size;
    }
```

Reply:
Comment above related to other PR's with kernels.

Reply:
@SlavikMIPT ping?
This blocks another PR: #7987

Reply:
@m-bronnikov 
> It's Okay to allocate memory without introduced memory allocator?

It is not the best way, but I think it is separate problem, because it requires refactoring of existing scratchpad related logic, BalyshevArtem is working on this problem.